### ISSUE TITLE
deleted lambda layer version in aws console, not cannot run "amplify push"

### ISSUE THREAD

**Issue by tomhirschfeld** [aws-amplify/amplify-cli#4948]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**

I deleted lambda layer version in aws console, and now I cannot run "amplify push", I am stuck trying to do a cloudformation rollback where it is stuck in the "UPDATE_ROLLBACK_FAILED" state, the cloudformation template references a lambda layer that no longer exists. How do I fix this issue? It seems I cannot create a lambda layer with a previously used version

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system

**To Reproduce**
Steps to reproduce the behavior or terminal output if applicable

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Mac/Windows/Ubuntu]
 - Node Version. You can use `node -v` to check the node version on your system

**Additional context**

**Comment by tomhirschfeld** [comment]
hey @jhockett I was unable to do that, I just ended up nuking my dev env and creating a new one, not ideal but was able to unblock myself. 

Still having the issue where the layer versions in CLI are out of sync w layer versions in console

**Comment by tomhirschfeld** [comment]
hey @jhockett I was unable to do that, I just ended up nuking my dev env and creating a new one, not ideal but was able to unblock myself.

Still having the issue where the layer versions in CLI are out of sync w layer versions in console

**Comment by cc-bojan** [comment]
@jhockett I'm also having a similar problem. After I created new Lambda Layer and wanted to associated that layer to existing Lambda function, I am not able to update CloudFormation stack anymore, nor I'm able to remove old Lambda Layer for that matter:

**Step: 1** (update _MyLambda_ function to not use the Layer)
```
⇒  amplify update function
? Select which capability you want to update: Lambda function (serverless function)
? Select the Lambda function you want to update MyLambda
? Do you want to update the Lambda function permissions to access other resources in this project? No
? Do you want to invoke this function on a recurring schedule? No
? Do you want to configure Lambda layers for this function? Yes
? Provide existing layers or select layers in this project to access from this function (pick up to 5): (Press <space> to select, <a> to toggle all, <i> to invert selection)
 ◯ Provide existing Lambda layer ARNs
❯◯ my-deprecated-lambda-layer
```
**Step: 2** (remove Lambda Layer)

```⇒  amplify remove function
? Choose the resource you would want to remove my-deprecated-lambda-layer (layer)

When you delete a layer version, you can no longer configure functions to use it.
However, any function that already uses the layer version continues to have access to it.
? Are you sure you want to delete the resource? This action deletes all files related to this resource from the backend directory. Yes
Resource cannot be removed because it has a dependency on another resource
Dependency: Lambda:MyLambda
Error: Resource cannot be removed because it has a dependency on another resource
    at /usr/local/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/remove-resource.js:128:31
    at Array.forEach (<anonymous>)
    at /usr/local/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/remove-resource.js:124:40
    at Array.forEach (<anonymous>)
    at deleteResourceFiles (/usr/local/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/remove-resource.js:122:22)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Object.executeAmplifyCommand (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/lib/index.js:193:5)
    at async executePluginModuleCommand (/usr/local/lib/node_modules/@aws-amplify/cli/lib/execution-manager.js:153:5)
    at async Object.executeCommand (/usr/local/lib/node_modules/@aws-amplify/cli/lib/execution-manager.js:35:9)
    at async Object.run (/usr/local/lib/node_modules/@aws-amplify/cli/lib/index.js:82:9)
An error occurred when removing the function resource
```

Even tried to manually remove dependency from `backend-config.json`:

```
   "MyLambda": {
      "build": true,
      "providerPlugin": "awscloudformation",
      "service": "Lambda",
      "dependsOn": [
        {
          "category": "function",
          "resourceName": "my-deprecated-lambda-layer",
          "attributes": [
            "Arn"
          ]
        }
      ]
    },
```  

However, when I try to push something using amplify cli, I'm getting this error:

<img width="1194" alt="1" src="https://user-images.githubusercontent.com/44865325/99144950-b5000b80-266a-11eb-9b38-4da9dfa5e28b.png">

This brakes my entire environment and I don't know what to do next.

**Comment by cellularatomaton** [comment]
@cc-bojan :  I have been struggling with the same issue for a week now and just resolved it.  Here is what I did:  Search your source code for the field `dependsOn[...]`.  I did this early on but it didn't work because the amplify entries in `.gitignore` were obscuring a number of the files in my IDE (VS Code).  Once I temporarily renamed `.gitignore` to `not.gitignore` I was able to see the dangling `dependsOn[...]` references.  Upon removing the problematic layer from those `dependsOn[...]` sections, I was able to successfully run `amplify remove function`.  Hope you are able to benefit from me pulling my hair out over the past week!
